azure-pipelines.yml

```diff
@@ -1,342 +1,424 @@
-      value: "62864d44-5da9-4ae9-89e7-0cf33942fa09"
-    ${{ if eq( parameters['env'], 'demo') }}:
-      value: "d025fece-ce99-4df2-b7a9-b649d3ff2060"      
-    ${{ if eq( parameters['env'], 'ptl') }}:
-      value: "1baf5470-1c3e-40d3-a6f7-74bfbce4b348"
-    ${{ if eq( parameters['env'], 'preview') }}:
-      value: "8b6ea922-0862-443e-af15-6056e1c9b9a4"
-    ${{ if eq( parameters['env'], 'aat') }}:
-      value: "96c274ce-846d-4e48-89a7-d528432298a7"
-    ${{ if eq( parameters['env'], 'prod') }}:
-      value: "8cbc6f36-7c56-4963-9d36-739db5d00b27"   
-
 
 stages:
-
-  - stage: PreChecks
-    displayName: 'PreChecks'
+  - stage: PreCheck
+    displayName: "PreChecks"
     jobs:
       - job: PreChecks
         steps:
           - template: steps/terraform-precheck.yaml@cnp-azuredevops-libraries
             parameters:
               keyvaultName: infra-vault-nonprod
-              keyvaultSecret: 'azure-devops-sp-token'
+              keyvaultSecret: "azure-devops-sp-token"
               serviceConnection: azurerm-sandbox
               projectName: ${{ variables.project }}
               cluster: ${{ parameters.cluster }}
-              environment: ${{ parameters.env }}
+              environment: sbox
               runManualStart: true
+  - ${{ each component in parameters.environment_components }}:
+      - stage: "CheckingClusters_${{ component.env }}"
+        displayName: "Checking Clusters for ${{ component.env }}"
+        dependsOn:
+          - ${{ if ne(component.env, 'sbox') }}:
+              - "Network_${{ component.dependsOn }}"
+              - "Managed_Identity_${{ component.dependsOn }}"
+              - "Aks_${{ component.dependsOn }}"
+              - "PipelineTests_${{ component.dependsOn }}"
+              - "BootStrapClusters_${{ component.dependsOn }}"
+          - ${{ else }}:
+              - "Precheck"
+        jobs:
+          - job: CheckingClusters
+            steps:
+              - checkout: self
+              - task: AzureCLI@2
+                displayName: "Checking available clusters"
+                inputs:
+                  scriptType: bash
+                  azureSubscription: ${{ component.serviceConnection }}
+                  scriptPath: "scripts/checking-clusters.sh"
+                name: check_clusters
+              - script: |
+                  echo "Cluster output: $(check_clusters.cluster_deploy)"
+                displayName: "Print cluster output"
+                name: set_cluster_output
+              - script: |
+                  echo "Cluster Parameter: ${{ parameters.cluster }}"
+                  if [ "${{ parameters.cluster }}" = "$(check_clusters.cluster_deploy)" ]; then
+                      echo "Input matches cluster"
+                  else
+                      echo "Input does not match cluster"
+                  fi
+                displayName: "Print cluster parameter and output"
+                name: compare_cluster_and_output
```

Comment:
Do we actually need this? I thought we're just using the 
```
              - task: AzureCLI@2
                displayName: "Checking available clusters"
```
task to set a var to reference in other stages, do we need to print it and then print if it matches or not?

Reply:
i thought it would be good to know but i remove the echo too on it

Reply:
Adds a lot of code into the build file for not much reason, will make things difficult to debug/follow along if it's repeating the same thing 3 times

Reply:
made it a little smaller